app/models/question.rb

```diff
@@ -2,12 +2,19 @@ class Question < ApplicationRecord
   before_create :set_seq
   belongs_to :questionnaire # each question belongs to a specific questionnaire
   has_many :answers, dependent: :destroy
-  
+  has_many :quiz_question_choices, class_name: 'QuizQuestionChoice', foreign_key: 'question_id'
+
   validates :seq, presence: true, numericality: true # sequence must be numeric
   validates :txt, length: { minimum: 0, allow_nil: false, message: "can't be nil" } # user must define text content for a question
   validates :question_type, presence: true # user must define type for a question
   validates :break_before, presence: true
 
+  QUESTION_TYPES = {
+    'MultipleChoiceCheckbox' => :calculate_score_for_checkbox_question,
+    'TrueFalse' => :calculate_score_for_truefalse_question,
```

Comment:
Method name is too complicated.  Couldn't there just be a **score** method implemented in Response, that could be overridden where necessary?

-Ed